src/response/chi0.jl

```diff
@@ -103,6 +104,47 @@ LinearAlgebra.ldiv!(y::T, P::FunctionPreconditioner, x) where {T} = P.preconditi
 LinearAlgebra.ldiv!(P::FunctionPreconditioner, x) = (x .= P.precondition!(similar(x), x))
 precondprep!(P::FunctionPreconditioner, ::Any) = P
 
+# custom cg to ensure that the iterates stay in Ran(R), taken from
+# https://github.com/JuliaLinearAlgebra/IterativeSolvers.jl/blob/master/src/cg.jl
+function cg(A, b, R=Identity();
```

Comment:
You might want to use multiple dispatch here instead of defining a new function.

Also this feels a lot like (a) too much code duplication with what `IterativeSolvers` already has and (b) something that breaks easily when `IterativeSolvers` refactors stuff. Are we really sure this is long-term the easiest and best way to solve this problem? Should we not better make a PR upstream?

Reply:
You mean a PR to IterativeSolvers.jl ?

Reply:
>  too much code duplication 
Yes I agree, but I can't figure out a shorter way of doing that (except a PR to IterativeSolvers)

Reply:
> You mean a PR to IterativeSolvers.jl ?

Yes, exactly.